api/src/opentrons/protocol_api/core/engine/module_core.py

```diff
@@ -175,138 +175,133 @@ class ThermocyclerModuleCore(ModuleCore, AbstractThermocyclerCore[LabwareCore]):
-        raise NotImplementedError("get_lid_temperature_status not implemented")
+        return self._sync_module_hardware.lid_temp_status  # type: ignore[no-any-return]
 
     def get_block_temperature(self) -> Optional[float]:
         """Get the thermocycler's current block temperature in °C."""
-        raise NotImplementedError("get_block_temperature not implemented")
+        return self._sync_module_hardware.temperature  # type: ignore[no-any-return]
 
     def get_block_target_temperature(self) -> Optional[float]:
         """Get the thermocycler's target block temperature in °C."""
-        raise NotImplementedError("get_block_target_temperature not implemented")
+        return self._sync_module_hardware.target  # type: ignore[no-any-return]
 
     def get_lid_temperature(self) -> Optional[float]:
         """Get the thermocycler's current lid temperature in °C."""
-        raise NotImplementedError("get_lid_temperature not implemented")
+        return self._sync_module_hardware.lid_temp  # type: ignore[no-any-return]
 
     def get_lid_target_temperature(self) -> Optional[float]:
         """Get the thermocycler's target lid temperature in °C."""
-        raise NotImplementedError("get_lid_target_temperature not implemented")
+        return self._sync_module_hardware.lid_target  # type: ignore[no-any-return]
 
     def get_ramp_rate(self) -> Optional[float]:
-        """Get the thermocycler's current rampe rate in °C/sec."""
-        raise NotImplementedError("get_ramp_rate not implemented")
+        """Get the thermocycler's current ramp rate in °C/sec."""
+        return self._sync_module_hardware.ramp_rate  # type: ignore[no-any-return]
 
     def get_hold_time(self) -> Optional[float]:
         """Get the remaining hold time in seconds."""
-        raise NotImplementedError("get_hold_time not implemented")
+        return self._sync_module_hardware.hold_time  # type: ignore[no-any-return]
 
     def get_total_cycle_count(self) -> Optional[int]:
         """Get number of repetitions for current set cycle."""
-        raise NotImplementedError("get_total_cycle_count not implemented")
+        return self._repetitions
 
     def get_current_cycle_index(self) -> Optional[int]:
         """Get index of the current set cycle repetition."""
-        raise NotImplementedError("get_current_cycle_index not implemented")
+        return None if self._repetitions is None else self._repetitions + 1
 
     def get_total_step_count(self) -> Optional[int]:
         """Get number of steps within the current cycle."""
-        raise NotImplementedError("get_total_step_count not implemented")
+        return self._step_count
 
     def get_current_step_index(self) -> Optional[int]:
         """Get the index of the current step within the current cycle."""
-        raise NotImplementedError("get_current_step_index not implemented")
+        return None if self._step_count is None else self._step_count + 1
```

Comment:
This looks like it's always going to return `len(steps) + 1`, which isn't right. It should be derived from the hardware API's understanding of the current step.

Something along the lines of:

```suggestion
        return (
            None
            if self._step_count is None
            else self._sync_module_hardware.current_step_index % self._step_count
        )
```

Don't trust me on that exact math; I recommend writing out a table and check it by hand, e.g.

- User-provided steps length
- User-provided repetitions count
- Total step length
- Actual 0-based total step index
- Reported 1-based hardware step index
- Desired user-facing step index